spinalcordtoolbox/gui/base.py

```diff
@@ -0,0 +1,311 @@
+        self.lb_warning.setAlignment(QtCore.Qt.AlignCenter)
+
+        parent.addWidget(self.lb_status)
+        parent.addWidget(self.lb_warning)
+        parent.addItem(
+            QtGui.QSpacerItem(20, 40, QtGui.QSizePolicy.Minimum,
+                              QtGui.QSizePolicy.Expanding))
+        message = getattr(self.params, 'init_message', '')
+        self.update_status(message)
+
+    def _init_footer(self, parent):
+        """
+
+        Parameters
+        ----------
+        parent : QtGui.QWidget
+            The widget / dialog that will host the footer layout
+
+        Returns
+        -------
+            The footer layout created
+        """
+        ctrl_layout = QtGui.QHBoxLayout()
+
+        self.btn_ok = QtGui.QPushButton('Save and Quit')
+        self.btn_undo = QtGui.QPushButton('Undo')
+        self.btn_help = QtGui.QPushButton('Help')
+
+        ctrl_layout.addItem(
+            QtGui.QSpacerItem(40, 20, QtGui.QSizePolicy.Expanding,
+                              QtGui.QSizePolicy.Minimum))
+        ctrl_layout.addWidget(self.btn_help)
+        ctrl_layout.addWidget(self.btn_undo)
+        ctrl_layout.addWidget(self.btn_ok)
+
+        self.btn_help.clicked.connect(self.on_help)
+        self.btn_undo.clicked.connect(self.on_undo)
+        self.btn_ok.clicked.connect(self.on_save_quit)
+
+        parent.addLayout(ctrl_layout)
+        return ctrl_layout
+
+    def on_save_quit(self):
+        self._controller.save()
+        self.close()
+
+    def on_undo(self):
+        try:
+            self._controller.undo()
+        except InvalidActionWarning as err:
+            self.update_warning(err.message)
+
+    def on_help(self):
+        webbrowser.open(self._help_web_address, new=0, autoraise=True)
+
+    def show(self):
+        """Override the base class show to fix a bug found in MAC"""
+        super(BaseDialog, self).show()
+        self.activateWindow()
+        self.raise_()
+
+    def update_status(self, msg):
+        """Print the message into the dialog's status widget and clear the warning widget
+
+        Parameters
+        ----------
+        msg : str  The message to display in the header of dialog
+        """
+        self.lb_status.setText(msg)
+        self.lb_warning.setText('')
+
+    def update_warning(self, msg):
+        """Print the message into the dialog's warning widget and clear the status widget
+
+        Parameters
+        ----------
+        msg : str  The message to display in the header of dialog
+        """
+        self.lb_warning.setText(msg)
+        self.lb_status.setText('')
+
+
+class BaseController(object):
+    orientation = None
+    _overlay_image = None
+    _dialog = None
+    position = None
+
+    def __init__(self, image, params, init_values=None):
+        self.image = image
+        self.params = params
+        self.points = []
+
+        if init_values:
+            self._overlay_image = init_values
+
+    def align_image(self):
```

Comment:
Shouldn't be called `align_image`. As I understand, it changes orientation and normalize the intensity (min and max).

Reply:
I'm having a hard time finding a good name for this function. Maybe `readjust_image`?

Reply:
how about `reformat_image`?
also, please add comment about what this function does